OptimizelySDK/OptimizelyUserContext.cs

```diff
@@ -58,8 +59,11 @@ public void SetAttribute(string key, object value)
                 Logger.Log(LogLevel.WARN, "Null attribute key.");
             }
             else
-            { 
-                UserAttributes.Add(key, value);
+            {
+                lock(mutex) 
+                { 
+                    UserAttributes[key] = value;
+                }
```

Comment:
Don't we also need control reading attributes under the same mutex lock? Need protect concurrent access for reading attributes from "decide" and updating attributes from "setAttribute".

Reply:
I think we should pass a copy to decide of attributes. Lock is fine for only writing.